_posts/2017-10-20-jbhc17-from-my-perspective.md

```diff
@@ -0,0 +1,95 @@
+But the majority of talks during the first day were not about that. I have to say, it got me worried a bit.
+It was interesting, but still. Little did I know that JB had a plan.
+He introduced a way of working in order to solve one problem: bottlenecks.
+And he did a great job. We learned how to find what slows us down and how to tackle this.
+He showed us how to work efficiently and what eventually matters - our clients, time to bringing features to them
+is the most important thing. The shorter the better, as that is all that matters.
+
+How?
+
+He started off with [Winston W. Royce's waterfall][waterfall] model, then, step by step,
+changed it into test first approach.
+From queueing theory, test first programming to TDD and BDD. He spoke a lot about throughput of our work,
+from raw material to final product. How it flows through the system, where the constraints lie.
+JB described how value stream mapping works, which is the technique used to discovering bottlenecks.
+You think you have a high efficiency in your team? 30-50%? Think again :)
+We did the exercise and it was ~6%. Not kidding.
+
+First day introduced a lot of very useful knowledge, and went by fast.
+It ended with a very nice dinner at a pizza restaurant. We had a chance to talk about anything with JB.
+You really need to meet this guy in person.
+
+## Second day
+
+At that day we split into two groups: programmers and product people.
+First group had two exercises to choose from. I picked the one where we were supposed to build a cash
+register. From very simple tasks (scan one product, show price) to advanced features.
+The reason for this was to practice TDD approach (don't mistake it with TFP).
+Why this? JB showed us a very simple diagram, a loop which is all about TDD: red - green - refactor.
+As for the refactor part, JB mentioned two most important things to do, which refactoring boils down to:
+ 1. Removing duplication
+ 1. Improving names
+
+It was a very nice experience. Step by step, when I was adding features, I saw how easy it was to add another one
+if code I produced was left in a good shape. It involved removal of duplication, introducing abstraction, providing
+better naming, etc. It felt great doing this.
+
+After the exercise, JB continued talking about the cost of introducing TDD: it is an investment, but such
+that pays of in long term (not that long!). As day went by JB told us also about few other, very interesting
+facts and figures backing up test first approach.
+
+This day ended with a really big surprise. **JBHC17 will happen again!** It will take place on 11-12 December.
+I already know I will go. It's one of those things you can't miss.
+
+## Final words
+
+JB is a very nice person, he's very smart and he's a real professional.
+During those two days I learned more about him, his personal life,
+how passionate he is in what he is doing. One can say - but he's being paid for what he does.
```

Comment:
"One can say - but he's being paid for what he does" -> "You could argue that he is getting paid for what he does"

Reply:
"You could argue that he is getting paid for what he does" - I guess this will sound like "I don't think he's getting paid", which is quite the opposite.

I will check it out tomorrow with Alex, our English specialist ;)

Reply:
How about "You could argue **because** he is getting paid for what he does"?

Reply:
IMHO, the sentence could be misunderstood as being somewhat derogative, as in "since we pay him, we can expect passion/quality".

I think the paragraph stands well on its own without the last sentence. My suggestion would be to drop the sentence entirely.

Reply:
@timoreimann sounds good to me.

Reply:
OK, will do so. Thanks @timoreimann @gilligan